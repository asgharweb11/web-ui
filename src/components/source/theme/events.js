import theme from "./theme"

export const Color = (item , param) => {
    switch(item){
        case "ligth" :
            return theme.ligth.text.grey;
        case "dark" :
            return theme.dark.text.grey;
    }
}