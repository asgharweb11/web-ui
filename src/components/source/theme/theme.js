export default {
    ligth : {
        main : "white",
        content : "repeating-linear-gradient(0deg , #f2efff , #f3f3ff)",
        primary : "#2196f3",
        secondary : "#ff80ab",
        error : "#f44336",
        warning : "#ff9800",
        info : "#2196f3",
        success : "#4caf50",
        text : {
            grey : "#313131",
            grey1 : "#656565",
            grey2 : "#909090",
            grey3 : "#808080",
            grey4 : "#9a9a9a",
            grey5 : "#c7c7c7",
            color : "#ffffff",
        },
        glass : {
            grey : "#000000a3",
            grey2 : "#00000066",
            grey3 : "#0000003b",
            grey4 : "#0000001c",
            grey5 : "#0000000d",
        },
        hover : {
            primary : "#246fab",
            secondary : "#ec5286",
            error : "#dc1709",
            warning : "#e08703",
            info : "#0d7ad2",
            success : "#237b26",
        }
    },  
    dark : {
        main : "#212121",
        content : "repeating-linear-gradient(0deg ,#333333 ,#333333)",
        primary : "#4dabf5",
        secondary : "#ff99bb",
        error : "#e57373",
        warning : "#ffb74d",
        info : "#64b5f6",
        success : "#81c784",
        text : {
            grey : "#ffffff",
            grey1 : "#ececec",
            grey2 : "#c7c7c7",
            grey3 : "#212121",
            grey4 : "#6b6b6b",
            grey5 : "#464646",
            color : "2b2b2b"
        },
        glass : {
            grey : "#ffffffb3",
            grey2 : "#ffffff87",
            grey3 : "#ffffff59",
            grey4 : "#ffffff2b",
            grey5 : "#ffffff0f",
        },
        hover : {
            primary : "#90ceff",
            secondary : "#ffb5ce",
            error : "#fdaaaa",
            warning : "#ffd393",
            info : "#a8d8ff",
            success : "#b0efb2",
        }

    },
    fontSize : {
        h1 : "40px",
        xl : "25px",
        lg : "20px",
        md : "16px",
        sm : "13px",
        xs : "10px",
    },
    px : {
        p1 : "0px",
        p2 : "1px",
        p3 : "5px",
        p4 : "10px",
        p5 : "15px",
        p6 : "20px",
    },
    media : {
        mobileS: '320px',
        mobileM: '375px',
        mobileL: '425px',
        tablet: '768px',
        tabletL : "991px",
        laptop: '1024px',
        laptopM: '1200px',
        laptopL: '1440px',
        desktop: '2560px'
    }

}