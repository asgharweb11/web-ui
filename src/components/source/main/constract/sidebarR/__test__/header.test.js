import React from "react"
import renderer from "react-test-renderer"
// -------- Context
import Context from "../../../../../../context.js"
// -------- Styled-ThemeProvider
import {ThemeProvider} from "styled-components"
import theme from "../../../../theme/theme"
//--------- Component
import Header from "../header"


test('Component : Constract/header.js', () => {
    const container = renderer.create(<Context>
        <ThemeProvider theme={theme}>
            <Header />
        </ThemeProvider>
    </Context>)

    let tree = container.toJSON();
    expect(tree).toMatchSnapshot()
})
