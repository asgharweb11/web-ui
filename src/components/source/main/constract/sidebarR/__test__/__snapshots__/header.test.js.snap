// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component : Constract/header.js 1`] = `
<div
  className="header"
>
  <img
    alt="sidebarR"
    className="picSidebar"
    src="/files/pic/detail/sidebar2.png"
  />
  <div
    className="detail"
  >
    <div
      className="profile"
    >
      <img
        alt="user"
        src="/files/pic/user/user.jpg"
      />
    </div>
    <div
      className="other"
    >
      <div
        className="icon"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-user fa-w-14 "
          data-icon="user"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <div
        className="icon"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-newspaper fa-w-18 "
          data-icon="newspaper"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M552 64H88c-13.255 0-24 10.745-24 24v8H24c-13.255 0-24 10.745-24 24v272c0 30.928 25.072 56 56 56h472c26.51 0 48-21.49 48-48V88c0-13.255-10.745-24-24-24zM56 400a8 8 0 0 1-8-8V144h16v248a8 8 0 0 1-8 8zm236-16H140c-6.627 0-12-5.373-12-12v-8c0-6.627 5.373-12 12-12h152c6.627 0 12 5.373 12 12v8c0 6.627-5.373 12-12 12zm208 0H348c-6.627 0-12-5.373-12-12v-8c0-6.627 5.373-12 12-12h152c6.627 0 12 5.373 12 12v8c0 6.627-5.373 12-12 12zm-208-96H140c-6.627 0-12-5.373-12-12v-8c0-6.627 5.373-12 12-12h152c6.627 0 12 5.373 12 12v8c0 6.627-5.373 12-12 12zm208 0H348c-6.627 0-12-5.373-12-12v-8c0-6.627 5.373-12 12-12h152c6.627 0 12 5.373 12 12v8c0 6.627-5.373 12-12 12zm0-96H140c-6.627 0-12-5.373-12-12v-40c0-6.627 5.373-12 12-12h360c6.627 0 12 5.373 12 12v40c0 6.627-5.373 12-12 12z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
